copy/multiple.go

```diff
@@ -140,6 +218,32 @@ func (c *copier) copyMultipleImages(ctx context.Context) (copiedManifest []byte,
 				UpdateSize:                  int64(len(updated.manifest)),
 				UpdateCompressionAlgorithms: updated.compressionAlgorithms,
 				UpdateMediaType:             updated.manifestMIMEType})
+		case instanceCopyClone:
+			logrus.Debugf("Replicating instance %s (%d/%d)", instance.sourceDigest, i+1, len(instanceCopyList))
+			c.Printf("Replicating image %s (%d/%d)\n", instance.sourceDigest, i+1, len(instanceCopyList))
+			unparsedInstance := image.UnparsedInstance(c.rawSource, &instanceCopyList[i].sourceDigest)
+			if instance.compressionVariant != nil {
+				// ---- THIS BLOCK IS A HACK and will not be merged till: https://github.com/containers/image/pull/2050
+				// copy options with new compression format
+				if c.options.DestinationCtx == nil {
+					c.options.DestinationCtx = &types.SystemContext{}
+				}
+				c.options.DestinationCtx.CompressionFormat = instance.compressionVariant
+			}
+			updated, err := c.copySingleImage(ctx, unparsedInstance, &instanceCopyList[i].sourceDigest, true)
+			if err != nil {
+				return nil, fmt.Errorf("replicating image %d/%d from manifest list: %w", i+1, len(instanceCopyList), err)
+			}
+			// Record the result of a possible conversion here.
+			instanceEdits = append(instanceEdits, internalManifest.ListEdit{
+				ListOperation:            internalManifest.ListOpAdd,
+				AddDigest:                updated.manifestDigest,
+				AddSize:                  int64(len(updated.manifest)),
+				AddMediaType:             updated.manifestMIMEType,
+				AddPlatform:              instance.cloneAddPlatform,
+				AddAnnotations:           instance.cloneAddAnnotations,
+				AddCompressionAlgorithms: instance.cloneAddCompressionAlgorithms,
```

Comment:
This can now  be `updated.compressionAlgorithms` — and that `instanceCopy` filed does not need to exist.